src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -1,6 +1,9 @@
 #include "common.h"
 
 ceph::mutex glock = ceph::make_mutex("glock");
+class SampleDedupWorkerThread;
+std::list<SampleDedupWorkerThread> threads;
+bool all_stop = false;
```

Comment:
Before moving onto your next work item, please submit a PR refactoring these globals into an automated variable declared in main().